### ISSUE TITLE
"Out of order sample from remote write" after agent restart

### ISSUE THREAD

**Issue by fatpat** [prometheus/prometheus#9867]
**What did you do?**
I'm setting up a prom/agent configuration. I'm using version `2.32.0-beta.0`.

I have:
- 1 prometheus server configured as normal and with remote write enable (`--enable-feature=remote-write-receiver`)
- 1 prometheus agent configured to scrape 19 targets and to remote write to the prometheus server

**What did you expect to see?**
metrics from the agent to be sent and be available from the server

**What did you see instead? Under which circumstances?**
when the agent directory is empty at agent start, it works correctly. But when I restart the agent without deleting the agent directory in between, I'm randomly having some errors as:

```
Nov 25 15:51:25 prometheus[1154732]: ts=2021-11-25T15:51:25.658Z caller=write_handler.go:57 level=error component=web msg="Out of order sample from remote write" err="out of order sample"
Nov 25 15:55:55 prometheus[1154732]: ts=2021-11-25T15:55:55.561Z caller=write_handler.go:57 level=error component=web msg="Out of order sample from remote write" err="duplicate sample for timestamp"
```

**Environment**
linux ubuntu focal

* System information:
```
Linux 5.11.0-27-generic x86_64
```

* Prometheus version:
```
prometheus, version 2.32.0-beta.0 (branch: HEAD, revision: c32725ba7873dbaa39c223410043430ffa5a26c0)
  build user:       root@da630543d231
  build date:       20211116-11:23:14
  go version:       go1.17.3
  platform:         linux/amd64
```

* Prometheus Server configuration file:
```
global:
  scrape_interval: 15s
  scrape_timeout: 10s
  evaluation_interval: 15s
  external_labels:
    environment: test
rule_files:
- /etc/prometheus/alerts/*.yml
- /etc/prometheus/aggregations/*.yml
```
* Agent configuration file:
```
global:
  scrape_interval: 15s
  scrape_timeout: 10s
  evaluation_interval: 15s
  external_labels:
    environment: test
scrape_configs:
- job_name: netdata
  honor_timestamps: true
  params:
    format:
    - prometheus_all_hosts
  scrape_interval: 15s
  scrape_timeout: 10s
  metrics_path: /api/v1/allmetrics
  scheme: http
  follow_redirects: true
  file_sd_configs:
  - files:
    - /etc/prometheus/targets/netdata.yml
    refresh_interval: 5m
- job_name: exporter2
  honor_timestamps: true
  scrape_interval: 15s
  scrape_timeout: 10s
  metrics_path: /metrics
  scheme: http
  follow_redirects: true
  file_sd_configs:
  - files:
    - /etc/prometheus/targets/exporter2.yml
    refresh_interval: 5m
- job_name: exporter1
  honor_timestamps: true
  scrape_interval: 15s
  scrape_timeout: 10s
  metrics_path: /metrics
  scheme: http
  follow_redirects: true
  file_sd_configs:
  - files:
    - /etc/prometheus/targets/exporter1.yml
    refresh_interval: 5m
- job_name: prometheus
  honor_timestamps: true
  scrape_interval: 15s
  scrape_timeout: 10s
  metrics_path: /metrics
  scheme: http
  follow_redirects: true
  file_sd_configs:
  - files:
    - /etc/prometheus/targets/prometheus.yml
    refresh_interval: 5m
remote_write:
- url: http://54.36.106.19:9090/api/v1/write
```

* Prometheus Logs:
```
Nov 25 15:51:25 prometheus[1154732]: ts=2021-11-25T15:51:25.576Z caller=write_handler.go:57 level=error component=web msg="Out of order sample from remote write" err="out of order sample"
Nov 25 15:51:25 prometheus[1154732]: ts=2021-11-25T15:51:25.646Z caller=write_handler.go:57 level=error component=web msg="Out of order sample from remote write" err="out of order sample"
Nov 25 15:51:25 prometheus[1154732]: ts=2021-11-25T15:51:25.658Z caller=write_handler.go:57 level=error component=web msg="Out of order sample from remote write" err="out of order sample"
Nov 25 15:55:55 prometheus[1154732]: ts=2021-11-25T15:55:55.561Z caller=write_handler.go:57 level=error component=web msg="Out of order sample from remote write" err="duplicate sample for timestamp"
Nov 25 15:56:55 prometheus[1154732]: ts=2021-11-25T15:56:55.561Z caller=write_handler.go:57 level=error component=web msg="Out of order sample from remote write" err="duplicate sample for timestamp"
```
* Agent Logs:
```
Nov 25 15:49:36 prometheus[48356]: ts=2021-11-25T15:49:36.352Z caller=dedupe.go:112 component=remote level=error remote_name=9dcd2a url=http://xxxx:9090/api/v1/write msg="non-recoverable error" count=500 exemplarCount=0 err="server returned HTTP status 400 Bad Request: out of order sample"
Nov 25 15:50:36 prometheus[48356]: ts=2021-11-25T15:50:36.371Z caller=dedupe.go:112 component=remote level=error remote_name=9dcd2a url=http://xxxx:9090/api/v1/write msg="non-recoverable error" count=500 exemplarCount=0 err="server returned HTTP status 400 Bad Request: out of order sample"

Nov 25 15:55:55 prometheus[50757]: ts=2021-11-25T15:55:55.567Z caller=dedupe.go:112 component=remote level=error remote_name=9dcd2a url=http://xxxx:9090/api/v1/write msg="non-recoverable error" count=500 exemplarCount=0 err="server returned HTTP status 400 Bad Request: duplicate sample for timestamp"
Nov 25 15:56:55 prometheus[50757]: ts=2021-11-25T15:56:55.567Z caller=dedupe.go:112 component=remote level=error remote_name=9dcd2a url=http://xxxx:9090/api/v1/write msg="non-recoverable error" count=500 exemplarCount=0 err="server returned HTTP status 400 Bad Request: duplicate sample for timestamp"
```

**Comment by fatpat** [comment]
fixed with 2.32.0-rc0

**Comment by ebarped** [comment]
Im having this issue (same logs both on prom-agent and prometheus) with the same setup (prom-agent sending via remote-write to a prometheus) in prometheus version:
```
prometheus, version 2.32.1 (branch: HEAD, revision: 41f1a8125e664985dd30674e5bdf6b683eff5d32)
  build user:       root@54b6dbd48b97
  build date:       20211217-22:08:06
  go version:       go1.17.5
  platform:         linux/amd64
```

Config:
```yaml
global:
  scrape_interval: 15s
  evaluation_interval: 15s
  external_labels:
    agent: "myagent"    

scrape_configs:
  - job_name: "agents"
    static_configs:
      - targets: ["localhost:9100"]
  - job_name: "processes"
    static_configs:
      - targets: ["localhost:9256"]

remote_write:
  - url: "<server>"
```
